src/services/kcur-floor-and-ceiling-service.ts

```diff
@@ -339,15 +345,18 @@ export const executeFloorAndCeilingService = async (
          */
         logMessage(serviceName, `The floor has been breached`);
         /**
-         * delta is how many kCUR we should be burning (selling) to bring the treasury value on par with
+         * delta is how many kCUR we should be buying to bring the reserve value on par with
          * the value of the total supply of kCUR.
          */
-        const delta = computeDelta(reserveStatus[0], kCurPrice, totalSupply, true);
+        const delta = await calculateRequiredkCur(
```

Comment:
can you please implement this logic into computeDelta, where it was before?  Minimize the changes...

Reply:
Done